### PR TITLE
feat: update api for mysql

### PR DESCRIPTION
**Pull request by kubeJocker**

### CODE REVIEW

File: addons/mysql/scripts/mysql-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+SERVICE_ID=$((${KB_POD_NAME##*-} + 1))
+if [ "${MYSQL_MAJOR}" = '5.7' ]; then
+  /scripts/docker-entrypoint-5.7.sh mysqld --server-id $SERVICE_ID --report-host ${REPORT_HOST} \
+    --ignore-db-dir=lost+found \
+    --plugin-load-add=rpl_semi_sync_master=semisync_master.so \
+    --plugin-load-add=rpl_semi_sync_slave=semisync_slave.so \
+    --plugin-load-add=audit_log=audit_log.so \
+    --log-bin=/var/lib/mysql/binlog/${KB_POD_NAME}-bin \
+    --skip-slave-start=$skip_slave_start
+else
```
Comment (nayutah) [inline addons/mysql/scripts/mysql-entrypoint.sh:12]: please add some annotations,  in fact it only works for 8.0

File: addons/mysql/templates/cmpdversion-mysql.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        init-jq: {{ .Values.image.registry | default "docker.io" }}/apecloud/jq:1.7.4
+        init-curl: {{ .Values.image.registry | default "docker.io" }}/apecloud/curl-static:7.67.0
+        init-xtrabackup: infracreate-registry.cn-zhangjiakou.cr.aliyuncs.com/apecloud/syncer:mysql
+        mysql: {{ .Values.image.registry | default "docker.io" }}/{{ .Values.image.repository }}:8.0.33
+    - name: {{ include "mysql.componentDefName" . }}-5.7.44
+      changes: mysql component version for orchestrator
+      serviceVersion: 5.7.44
+      images:
+        init-data: {{ .Values.image.registry | default "docker.io" }}/apecloud/mysql_audit_log:5.7.44
+        init-syncer: infracreate-registry.cn-zhangjiakou.cr.aliyuncs.com/apecloud/syncer:latest
```
Comment (nayutah) [inline addons/mysql/templates/cmpdversion-mysql.yaml:40]: keep image registry settings same with init-data

File: addons/mysql/scripts/mysql-entrypoint.sh (hunk 1 above)
Comment (kubeJocker) [reply addons/mysql/scripts/mysql-entrypoint.sh]: done

File: addons/mysql/templates/cmpdversion-mysql.yaml (hunk 2 above)
Comment (kubeJocker) [reply addons/mysql/templates/cmpdversion-mysql.yaml]: done

**Review by nayutah — APPROVED** [review]

**Comment by nayutah** [comment]
pls resolve the conflicts

File: addons/mysql/docker/tools/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
+FROM docker.io/netdata/wget:latest as build
```
Comment (shanshanying) [inline addons/mysql/docker/tools/Dockerfile:1]: move Dockerfile to an independant repo.

File: addons/mysql/templates/_helpers.tpl [hunk 4]
```diff
@@ -123,4 +123,295 @@ app.kubernetes.io/instance: {{ .Release.Name }}
 
 {{- define "mysql.imagePullPolicy" -}}
 {{ default "IfNotPresent" .Values.image.pullPolicy }}
+{{- end }}
+
+{{- define "mysql.spec.common" -}}
```
Comment (shanshanying) [inline addons/mysql/templates/_helpers.tpl:128]: seems this template is used in both cmpd 5.7 and cmpd 8.0, udpate service version accordingly.

File: addons/mysql/templates/_helpers.tpl [hunk 5]
```diff
@@ -123,4 +123,295 @@ app.kubernetes.io/instance: {{ .Release.Name }}
...
+  volumeMounts:
+    - mountPath: /kubeblocks
+      name: kubeblocks
+{{- end }}
+
+{{- define "mysql-orc.spec.common"}}
+labels:
+  kubeblocks.io/ready-without-primary: "true"
+provider: kubeblocks
+description: mysql 5.7 component definition for Kubernetes
```
Comment (shanshanying) [inline addons/mysql/templates/_helpers.tpl:218]: dito. check service version, cofigs, scripts, and other.  as this template is used for both cmpd 5.7 and cmpd 8.0

File: addons/mysql/templates/_helpers.tpl [hunk 6]
```diff
@@ -123,4 +123,295 @@ app.kubernetes.io/instance: {{ .Release.Name }}
...
+    serviceName: mysql
+    podService: true
+    spec:
+      ports:
+        - name: mysql
+          port: 3306
+          targetPort: mysql
+configs:
+  - name: mysql-replication-config
+    templateRef: mysql-5.7-config-template
```
Comment (shanshanying) [inline addons/mysql/templates/_helpers.tpl:246]: check configs.

File: addons/mysql/templates/_helpers.tpl [hunk 7]
```diff
@@ -123,4 +123,295 @@ app.kubernetes.io/instance: {{ .Release.Name }}
...
+    constraintRef: mysql-config-constraints
+    volumeName: mysql-config
+    namespace: {{ .Release.Namespace }}
+scripts:
+  - name: mysql-scripts
```
Comment (shanshanying) [inline addons/mysql/templates/_helpers.tpl:251]: check scripts. better  to used two configmap to mount configs  5.7 and 8.0, one for each.

File: addons/mysql/templates/_helpers.tpl [hunk 8]
```diff
@@ -123,4 +123,295 @@ app.kubernetes.io/instance: {{ .Release.Name }}
...
+    serviceable: true
+    writable: true
+  - name: secondary
+    serviceable: true
+    writable: false
+lifecycleActions:
+  roleProbe:
+    builtinHandler: custom
+    customHandler:
+      image: {{ .Values.image.registry | default "docker.io" }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}
```
Comment (shanshanying) [inline addons/mysql/templates/_helpers.tpl:279]: remove image when not necessary ( esp for lorry)

**Comment by shanshanying** [comment]
pls start with mysql 8.0.33 with orc first:
1. create an orc cluster
2. create an mysql 8.0.33 cluster with orc

File: addons/mysql/docker/tools/Dockerfile (hunk 3 above)
Comment (kubeJocker) [reply addons/mysql/docker/tools/Dockerfile]: done

**Comment by kubeJocker** [comment]
> pls start with mysql 8.0.33 with orc first:
> 
> 1. create an orc cluster
> 2. create an mysql 8.0.33 cluster with orc
> 3. memberLeave does not work as expected
> 4. per-pod svc are not recycled on hscale-in.

all solved

**Pull request merged**